### DISCUSSION TITLE
Executing JS scripts within events

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#783] [Ideas]
There's currently no existing capabilities for running a JS script from within an event tag.

I'm currently unsure what kind of API should exist for something like this though.

For example, perhaps it would look something like this?

```python
@component
def my_component():
    return html.button(
        { "onClick": "$('.sidebar').toggle()" },
        "This is a button!",
    )
```

In this scenario, providing a `str` within `onClick` would execute a one-time `eval( ... )` on the client side.

This discussion will be converted to an issue after debating the API.

**Comment by rmorshea** [comment]
Can this be achieved with an adjacent `script` element that binds an onClick handler to the button?

**Reply by Archmonger** [reply]
Yes - But in my opinion that looks and feels quite awkward.

I think a string parameter for `onClick` (as seen above) feels intuitive. Especially since when writing raw HTML, one can typically do things such as
```html
<button onclick="myFunction()">Click me</button>
```

**Comment by Archmonger** [comment]
Extend the VDOM spec to allow for the following to occur
- String based event handlers
- `idom.core.event` based event handlers, that can access the serialized return attributes of some arbitrary javascript

**Reply by numpde** [reply]
Would it be possible to refer to idom components from within the JS string not via CSS (jQuery) but by their Python variable or idom-id? 

Does somehting meaningful happen for `{ "onClick": (lambda e: html.script(...)) }`? I think I could let the script write to a textarea-component and readout later / onChange?

**Comment by Archmonger** [comment]
Issue created
- https://github.com/reactive-python/reactpy/issues/1261
